hydra-node/exe/hydra-node/Main.hs

```diff
@@ -110,6 +126,29 @@ main = do
 
     action (Ledger.cardanoLedger globals ledgerEnv)
 
+  -- check if hydra-node parameters are matching with the hydra-node state.
+  -- REVIEW: Should we also check against items we receive on-chain here? e.g. 'OpenThreadOutput' if in Open state?
+  checkParamsAgainstExistingState :: HeadState Ledger.Tx -> Environment -> [String]
+  checkParamsAgainstExistingState hs env =
+    case hs of
+      Idle _ -> []
+      Initial InitialState{parameters} -> validateParameters "InitialState: " parameters
+      Open OpenState{parameters} -> validateParameters "OpenState: " parameters
+      Closed ClosedState{parameters} -> validateParameters "ClosedState: " parameters
+   where
+    validateParameters st params =
+      let res = flip execState [] $ do
```

Comment:
I think an applicative is more suitable here.

Why not use Validation instead?

Reply:
The code changed a bit but we could use validation package. The only thing I see bad is we end up depending on it but that should not be that bad after all.

Reply:
On second thought I would not do it right now since I don't think we will benefit much from using `validation`. There are just two cases we check here. Is that fine by you @ffakenz ?